### DISCUSSION TITLE
Echo listener with custom broadcastAs() not working

### DISCUSSION THREAD

**Discussion by AdvancedStyle** [livewire/livewire#3137] [7. Bugs (archived)]
### Description
If you set a custom name for your event using `broadcastAs()` then the `echo:{channel},{event}` listener does not work.

### Exact steps to reproduce
1) Make an event
2) Change the name of the event using broadcastAs(), for example:

```
public function broadcastAs(){
        return 'myeventname';
    }
```

3) Setup listener in your Livewire component, for example:

```
protected $listeners = ["echo:channelname,myeventname" => 'doSomething'];
```

It does not work and `doSomething` is never triggered.

### Workaround

Remove your custom event name from the event and use the default only, it now works.

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: 2.3.1
- Laravel version: 8.13.0
- Browser: Chrome

**Comment by MiracleInNutShell** [comment]
Hey try adding dot(.) Before your event name like this "echo:channel,.event"

**Comment by RhysLees** [comment]
> Hey try adding dot(.) Before your event name like this "echo:channel,.event"

~~This worked, thank you, but why does it not say this in the docs. the docs say to do it without the . so it don't really make a whole lot of sense.~~

~~are the docks incorrect or is there a big in livewire its self?~~

Edit: ignore me, its just me that has been doing it wrong the whole time, i didnt realise that livewire auto prepends App\Events\.

How has it taken me a year to figure this out?

**Reply by Tiththa** [reply]
Hi, Mind if you could share the workaround ? Thanks

**Reply by RhysLees** [reply]
So for your Event set your broadcastOn to what ever you want, eg

```PHP
public function broadcastOn()
    {
        return new PrivateChannel('command-log');
    }
```

Then in your Livewire Component set your lister as follows:

```PHP
public function getListeners()
    {
        return [
            "echo-private:command-log.{$this->guildID},BotCommandLog" => 'logUpdate',
        ];
    }
```

The reason you do this is because livewire prefixes `app\Events\` here: `"echo-private:command-log.{$this->guildID},<app\Events\>BotCommandLog" => 'logUpdate',` Which is the path to Event.

So this means if you have an event in a folder e.g `app\Events\Bot\BotCommandLog` then you would do 
```PHP
 "echo-private:command-log.{$this->guildID},Bot\BotCommandLog" => 'logUpdate',
```


Then your function in the Livewire component that handles the data

```PHP
public function logUpdate($data)
    {
        array_unshift($this->logs, $data['log']);
        if(count($this->logs) > 10){
            array_pop($this->logs);
        }
    }
```

I think it should explain it in the docs but just follow exactly what they do in the docs [here](https://laravel-livewire.com/docs/2.x/laravel-echo).


Hope this helps

**Reply by RhysLees** [reply]
the above also work for `protected $listeners =` i just prefer to use the function instead.

and to change from private to public channels change
```PHP
 "echo-private:command-log.{$this->guildID},Bot\BotCommandLog" => 'logUpdate',
```
to
```PHP
 "echo:command-log.{$this->guildID},,Bot\BotCommandLog" => 'logUpdate',
```

NOTE: for private channels you need to setup the channel route in `routes\channels.php` as follows:

```PHP
Broadcast::channel('command-log.{id}', function ($user, $id) {
    return $user->currentTeam->bot->guildID == $id;
});
```
This has to return true for the echo private channel to be able to connect, if you return false then it will fail with 403 I believe.
This is where you do your auth checks for the channels more can be found [here](https://laravel.com/docs/8.x/broadcasting#defining-authorization-callbacks).

**Reply by RhysLees** [reply]
If you need any more help reach out to me on discord NanoCellMusic#6969

**Reply by Loots-it** [reply]
I think an update to the docs would be nice. I had no idea about the dot in front of the event name.

**Reply by Tiththa** [reply]
Thanks for the update, just seen the notification @NanoCellWebDesign

**Reply by Wulfheart** [reply]
@Loots-it https://github.com/livewire/docs/pull/315
I made a PR some time ago but @calebporzio didn’t want to merge. Maybe show it some love.

**Reply by RhysLees** [reply]
> I think an update to the docs would be nice. I had no idea about the dot in front of the event name.

It does say this in the laravel echo docs but it would be nice to be included in the livewire docs aswell
